### PR TITLE
plot options support for scatter viewers/layers

### PR DESCRIPTION
**Pull request by kecnry**
### Description

This pull request adds hooks to glue state entries for scatter viewers and layers.  These currently are not used by jdaviz, but may in the future, and are not easy to implement downstream separately.  Because they are not used in any of the configs here, it is difficult to get coverage for the non-covered lines, but suggestions are welcomed!

NOTE: some attribute names may change if/when plotting lines in scatter viewers is exposed by glue-jupyter.

### Change log entry

- [x] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

File: jdaviz/configs/default/plugins/plot_options/plot_options.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/plot_options/plot_options.py:291]: this is what necessitates most of the changes to `template_mixin`, we want to handle two differently-named glue state attributes (`visible` for profile viewers and `line_visible` for scatter viewers) into the same plot options object.

File: jdaviz/configs/default/plugins/plot_options/plot_options.py
Comment (kecnry) [inline jdaviz/configs/default/plugins/plot_options/plot_options.py:339]: Even if we don't wait for the updates to glue-jupyter to support lines in scatter viewers, we might want to at least understand the behavior and what we want these to be called for the user-api.  Ultimately jdaviz doesn't use any of these, or expose any of these to the user, so making incremental changes to these wouldn't be a problem (just will need matching updates downstream to `lcviz`).

**Review by rosteen — APPROVED** [review]
I confirmed that this makes the appropriate toggles appear in the plot options for the ScatterViewer in `lcviz`, although showing the line does not currently work (expected). We discussed offline hiding that toggle until the required `glue-jupyter` work is done to make it actually work - after further thought I think that's the way to go. Approving now anyway, as that should be a small change (and I'm not adamant about it if you end up wanting to leave it as-is).

**Comment by kecnry** [comment]
Holding off on merge until the upstream PR gets in (or doesn't, in which case it would need to be implemented here instead of plot options support dropped for now and deferred).

**Review by bmorris3 — APPROVED** [review]
Looks good! 

I wondered if we have a use case when this might be useful in jdaviz (separate from LCviz), because if there is no use case, maybe it should live in LCviz. But I expect this will show up in another viz sometime, so I approve.

**Comment by kecnry** [comment]
> I support the option to remove marker_. Less typing without more ambiguity here, I think.

Agreed, we won't really know that until its added upstream in glue-jupyter, but since none of the jdaviz configs currently expose these as public user API, we can change them going forward without having to worry about backwards compatibility.

**Pull request merged**
